HM3CoreKill/HM3CoreKill/ck/HM3InGameTools.cpp

```diff
@@ -505,6 +607,113 @@ namespace ck
 		ImGui::End();
 	}
 
+	void HM3InGameTools::showInventoryEditorWindow(bool* pOpen)
+	{
+		auto sys = ioi::hm3::getGlacierInterface<ioi::hm3::ZSysInterfaceWintel>(ioi::hm3::SysInterface);
+		auto gameData = ioi::hm3::getGlacierInterface<ioi::hm3::ZHM3GameData>(ioi::hm3::GameData);
+		if (!gameData || !gameData->m_LevelControl)
+			return;
+
+		auto hitman3 = gameData->m_Hitman3;
+		auto engineDB = sys->m_engineDataBase;
+
+		ImGui::SetNextWindowSize(ImVec2(750, 400), ImGuiCond_FirstUseEver);
+		if (ImGui::Begin("Inventory editor", pOpen))
+		{
+			ImGui::Separator();
+			// left
+			static int selected = 0;
+			ImGui::BeginChild("left pane", ImVec2(150, 0), true);
+			for (int i = 0; i < gameData->m_ActorsInPoolCount; i++)
+			{
+				char label[128];
+				sprintf(label, "Actor #%.2d", i + 1);
+				if (ImGui::Selectable(label, selected == i))
+					selected = i;
+			}
+			ImGui::EndChild();
+			ImGui::SameLine();
+
+			// right
+			if (selected >= 0)
+			{
+				ioi::hm3::ZHM3Actor* currentActor = gameData->m_ActorsPool[selected];
+
+				ImGui::BeginGroup();
+				ImGui::BeginChild("item view", ImVec2(0, -ImGui::GetFrameHeightWithSpacing())); // Leave room for 1 line below us
+
+				ImGui::TextColored(ImVec4(0.f, 1.f, 0.f, 1.f), "INVENTORY:");
+				ImGui::Separator();
+
+				auto inventory = reinterpret_cast<ioi::hm3::CInventory*>(currentActor->getComponent("Inventory"));
+				if (inventory)
+				{
+					ioi::REFTAB32* inventoryREFTAB32 = inventory->getREFTAB32();
```

Comment:
Here we have a few bugs: 

1. If an actor has an item in their hands we should remove it from its hands before removing the item from inventory.
2. We can not remove any random item (removes only top item). For fix this we can remove all items from inventory and restore any that items who should be saved.

Reply:
Will be fixed in future PR.